src/main/java/io/jenkins/plugins/gitlabserverconfig/servers/GitLabServer.java

```diff
@@ -120,9 +122,17 @@ public class GitLabServer extends AbstractDescribableImpl<GitLabServer> {
     private String hooksRootUrl;
 
     /**
-     * The secret token used while setting up hook url in the GitLab server
+     * The credentials id of the secret token used while setting up hook url in the GitLab server
      */
-    private Secret secretToken;
+    @NonNull
+    private String secretTokenCredentialsId;
```

Comment:
Could consider calling it `credentialsId` but can also leave it as is.

Reply:
`credentialsId` is already used for the PAT credentials ID => https://github.com/jenkinsci/gitlab-branch-source-plugin/blob/69d3ba371450a9d0ff28fb917ffa90631f5a7f94/src/main/java/io/jenkins/plugins/gitlabserverconfig/servers/GitLabServer.java#L111-L116 

So I had to pick a different name, but I'm open for better names.

Reply:
how about `webhookSecretCredentialsId` 🤔 making it descriptive? 😅

Reply:
:+1: